packages/chain-adapters/src/api.ts

```diff
@@ -153,5 +153,5 @@ export interface ChainAdapter {
 
   broadcastTransaction(hex: string): Promise<string>
 
-  validateAddress(address: string): Promise<ValidAddressResult>
+  validateAddress(address: string): ValidAddressResult
```

Comment:
This intentionally returns a Promise even though it could be implemented synchronously at the moment in order to maintain implementation flexibility.

Reply:
Makes sense! I'll revert it.